src/modules/Dashboard/features/RespondentData/RespondentDataSummary/Report/Charts/BarChart/BarChart.tsx

```diff
@@ -30,5 +80,29 @@ export const BarChart = ({ chartData }: BarChartProps) => {
     datasets: getDatasets(chartData),
   };
 
-  return <Bar plugins={[legendMargin]} options={getOptions(chartData)} data={data} />;
+  const renderChart = useMemo(
+    () => (
+      <Bar
```

Comment:
why do you need to apply useMemo and block rerender for <Bar>? Could it lead to bugs if `data` will be updated?

Reply:
To prevent redundant and unnecessary chart rerenders and update only if the date has changed